src/components/templates/ConversationSpace/ConversationSpace.tsx

```diff
@@ -27,13 +30,25 @@ export const ConversationSpace: React.FunctionComponent = () => {
   const venue = useSelector(currentVenueSelectorData);
   const { recentVenueUsers } = useRecentVenueUsers({ venueName: venue?.name });
 
+  const { isShown: isChecked, toggle: toggleTables } = useShowHide();
+
+  const defaultTables = venue?.config?.tables ?? TABLES;
+  const { availableTables } = useTables({ defaultTables });
+
+  const [tables, setTables] = useState(defaultTables);
+
+  useEffect(() => setTables(defaultTables), [defaultTables]);
```

Comment:
useTables + useState + useEffect + onToggleTables -> the logic seem to be overcomplicated.

```ts
  const { isShown: areNotAvailableTablesShown, toggle: toggleNotAvailableTables } = useShowHide();

  const tablesToShow = useMemo(() => (venue?.config?.tables ?? TABLES).filter(table => {
    return areNotAvailableTablesShown || !(isLockedTable(table) || isFullTable(table))
  })), [...])
```
the names of varialbes can be better I suppose.

then you can remove `onToggleTables`

and in `<TableConrolBar handleChecked={toggleNotAvailableTables} .../>`

Reply:
that's just a draft solution.

same for JazzBar (and consider refactoring and moving the logic to some utils/table to reuse it)

Reply:
Thank you for help!
fixed in e8298bec7e35a4e40e7175b130015db2080f52e5